frontend/pages/configurator.vue

```diff
@@ -0,0 +1,1161 @@
+  try {
+    const token = localStorage.getItem("token");
+    const response = await $fetch(`${loadTestEndpoint}${inputTestId.value}`, {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    if (response && response.item_configs) {
+      testItems.value = response.item_configs;
+      selectedItems.value = response.item_configs.map((item) => item.id);
+      loadedTestId.value = response.id;
+      loadedTestName.value = response.name;
+      isLoadModalOpen.value = false;
+
+      triggerAlert("success", "Sehtest erfolgreich geladen!");
+    } else {
+      triggerAlert("error", "Fehler. Ungültige Antwortstruktur.");
+    }
+  } catch (error) {
+    if (error.response?.status === 404) {
+      triggerAlert("error", "Fehler: Dieser Sehtest existiert nicht");
+    } else {
+      triggerAlert("error", "Fehler beim Laden der Konfiguration.");
+    }
+  }
+};
+
+const resetModalFields = () => {
+  circleColor.value = "#ffcc00";
+  triangleColor.value = "#3333ff";
+  circleSize.value = 150;
+  triangleSize.value = 50;
+  modalOrientation.value = "N";
+};
+
+const openItemModal = (index = null) => {
+  if (index !== null) {
+    // Bearbeiten-Modus
+    const item = testItems.value[index];
+    circleColor.value = item.circle_color;
+    triangleColor.value = item.triangle_color;
+    circleSize.value = item.circle_size;
+    triangleSize.value = item.triangle_size;
+    modalOrientation.value = item.orientation;
+    isEditing.value = true;
+    editingIndex.value = index;
+  } else {
+    // Neu-Modus
+    resetModalFields(); // setzt die Felder zurück
+    isEditing.value = false;
+  }
+  isItemModalOpen.value = true;
+};
+
+const loadTestById = async (testId) => {
+  try {
+    const token = localStorage.getItem("token");
+    const response = await $fetch(`${loadTestEndpoint}${testId}`, {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    if (response && response.item_configs) {
+      testItems.value = response.item_configs;
+      selectedItems.value = response.item_configs.map((item) => item.id);
+      loadedTestId.value = response.id;
+      loadedTestName.value = response.name;
+      isLoadModalOpen.value = false;
+
+      triggerAlert("success", "Sehtest erfolgreich geladen!");
+    } else {
+      triggerAlert("error", "Fehler. Ungültige Antwortstruktur.");
+    }
+  } catch (error) {
+    triggerAlert("error", "Fehler beim Laden des Sehtests.");
+  }
+};
+
+const duplicateItem = (index) => {
+  const item = testItems.value[index];
+  const newItem = {
+    ...item,
+    id: crypto.randomUUID(), // Neue temporäre ID setzen, um die Einzigartigkeit sicherzustellen
+    isUnsaved: true, // Markiere das Item als ungespeichert
+    createdByUser: true, // Markiere es als durch den Nutzer erstellt
+  };
+
+  testItems.value.push(newItem); // Neues Item zu `testItems` hinzufügen
+};
+
+const deleteItem = (index) => {
```

Comment:
I'm not sure if I'm missing anything, but I think the item configs are deleted in the backend, but that should be the case, or what do you think?

Reply:
Im not quite sure what you mean... I think right now they actually wont be deleted in the database. Do you meant to say that? If so, I can defenitly change that if theres an endpoint for item config erasion. I believe the Item configs should be deleted in the DB as long as the user is allowed to do that.

Reply:
Yes, that is what I mean :) There is a DELETE endpoint that works in the same way as the DELETE endpoint for the test_config.

Reply:
I added the function, it should work now. However at one loaded test config which was created by myself i got an error message while trying to delete the item config. It was an backend error, i sent you the whole error on discord.